indico/modules/events/registration/controllers/__init__.py

```diff
@@ -36,3 +40,32 @@ def _checkParams(self):
                         .find(id=request.view_args['reg_form_id'], is_deleted=False)
                         .options(defaultload('form_items').joinedload('children').joinedload('current_data'))
                         .one())
+
+
+class RegistrationEditMixin(object):
+
+    def _process(self):
+        form = make_registration_form(self.regform, management=self.management, registration=self.registration)()
+
+        if form.validate_on_submit():
+            modify_registration(self.registration, form.data, self.event)
+            return redirect(self.success_url)
+        elif form.is_submitted():
+            # not very pretty but usually this never happens thanks to client-side validation
+            for error in form.error_list:
+                flash(error, 'error')
+
+        registration_data = {r.field_data.field.html_field_name: camelize_keys(r.data) for r in self.registration.data}
+        section_data = camelize_keys(get_event_section_data(
+            self.regform, management=self.management, registration=self.registration))
+
+        registration_data.update({
+            'paid': self.registration.state == RegistrationState.complete,
```

Comment:
I was going to suggest just that. I'd call it `is_unpaid` since `is_paid` is a bit confusing for a pending one.